src/orange/OrangeTrackView.hh

```diff
@@ -492,47 +537,100 @@ CELER_FUNCTION void OrangeTrackView::cross_boundary()
     CELER_EXPECT(this->is_on_boundary());
     CELER_EXPECT(!this->has_next_step());
 
-    auto lsa = this->make_lsa();
+    auto sl = states_.surface_level[track_slot_];
+    auto lsa = this->make_lsa(sl);
 
     if (CELER_UNLIKELY(lsa.boundary() == BoundaryResult::reentrant))
     {
+        CELER_ASSERT(sl.unchecked_get() == 0);
```

Comment:
Yes this was intentional. I am not sure I understand the use-case. Is this something specific to tracking through a B-field?

Reply:
Yes, it's something you only see with charged particles and not in neutron/gamma transport. Basically, a step can be geometry-limited (and move to the boundary), but in the process of moving that step, the magnetic field could change its direction so that instead of exiting it heads back into the cell it was about to leave. Similarly, "multiple scattering" happens along every step for charged particles, and that changes the direction at the end of the step. So a track that's "on the boundary" about to leave the cell ends up bouncing back in without actually leaving.

Reply:
If this is deliberately not implemented rather than an assertion for a normal code condition, also write like this:
```suggestion
        if (sl != LevelId{0})
        {
            CELER_NOT_IMPLEMENTED("reentrant surfaces with multilevel ORANGE geometry");
        }
```